internal/cmd/controller/reconciler/bundle_controller.go

```diff
@@ -499,33 +496,26 @@ func (r *BundleReconciler) updateStatus(ctx context.Context, orig *fleet.Bundle,
 	return nil
 }
 
-func findNotTargetedBundleDeployments(ctx context.Context, c client.Client, bundleId types.NamespacedName, targets []*target.Target) ([]types.NamespacedName, error) {
```

Comment:
@0xavi0 @manno I'm afraid that this implementation added at https://github.com/rancher/fleet/pull/3509/files#diff-38f599d4b6f84abae0fd9c6582e85b27f3f8a52774a565742e207ef52302274e completely overlaps with the changes I was proposing.
Nonetheless, I believe my proposal is a bit cleaner? (since it does not need to find again which BundleDeployments should be kept, but instead keeps them from the "creation phase").

Let me know if you think otherwise and I can trim this PR to only the improvements in finalizers.